From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: GrieferPig <grieferpig@163.com>
Date: Mon, 24 Aug 2026 11:15:00 +0800
Subject: [PATCH] linux: enable two pthread benchmark contexts

Exercise both ESP32-S31 Linux harts in the packaged CoreMark binary.  Keep the
change in the BR2_EXTERNAL patch overlay so the Buildroot submodule can remain
an unmodified upstream checkout.

Signed-off-by: GrieferPig <grieferpig@163.com>
---
 linux/core_portme.mak | 1 +
 1 file changed, 1 insertion(+)

diff --git a/linux/core_portme.mak b/linux/core_portme.mak
index 1598253..e1fe1fc 100755
--- a/linux/core_portme.mak
+++ b/linux/core_portme.mak
@@ -7,6 +7,7 @@ CC = gcc
 # Flag: CFLAGS
 #	Use this flag to define compiler options. Note, you can add compiler options from the command line using XCFLAGS="other flags"
 PORT_CFLAGS = -O2
+XCFLAGS += -DMULTITHREAD=2 -DUSE_PTHREAD=1 -pthread
 FLAGS_STR = "$(PORT_CFLAGS) $(XCFLAGS) $(XLFLAGS) $(LFLAGS_END)"
 CFLAGS = $(PORT_CFLAGS) -I$(PORT_DIR) -I. -DFLAGS_STR=\"$(FLAGS_STR)\"
 #Flag: LFLAGS_END
-- 
2.43.0

